### PR TITLE
Updates on Generated Code and New Tests added in .Net SDK

### PR DESCRIPTION
**Pull request by poadhikagithub**

### CODE REVIEW

File: src/SDKs/HealthcareApis/HealthcareApis.Tests/Helpers/HealthcareApisManagementTestUtilities.cs [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        }
+
+        public static ServicesPatchDescription GetServicePatchDescription()
+        {
+            var servicePatchDescription = new ServicesPatchDescription(UpdateTags);
+            return servicePatchDescription;
+
+        }
+
+        public static ServicesDescription GetServiceDescriptionWithProperties()
```
Comment (YazanMSFT) [inline src/SDKs/HealthcareApis/HealthcareApis.Tests/Helpers/HealthcareApisManagementTestUtilities.cs:79]: This isn't being used

File: src/SDKs/HealthcareApis/HealthcareApis.Tests/Helpers/HealthcareApisManagementTestUtilities.cs [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+                healthcareApisManagementClient = context.GetServiceClient<HealthcareApisManagementClient>(handlers: handler);
+            }
+            return healthcareApisManagementClient;
+        }
+
+        public static ServicesDescription GetServiceDescription()
+        {
+            var serviceDescription = new ServicesDescription(DefaultLocation);
+            return serviceDescription;
+
```
Comment (YazanMSFT) [inline src/SDKs/HealthcareApis/HealthcareApis.Tests/Helpers/HealthcareApisManagementTestUtilities.cs]: Nit: line

File: src/SDKs/HealthcareApis/HealthcareApis.Tests/Helpers/HealthcareApisManagementTestUtilities.cs [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        }
+
+        public static ServicesPatchDescription GetServicePatchDescription()
+        {
+            var servicePatchDescription = new ServicesPatchDescription(UpdateTags);
+            return servicePatchDescription;
+
```
Comment (YazanMSFT) [inline src/SDKs/HealthcareApis/HealthcareApis.Tests/Helpers/HealthcareApisManagementTestUtilities.cs]: Same here

File: src/SDKs/HealthcareApis/HealthcareApis.Tests/Helpers/HealthcareApisManagementTestUtilities.cs [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+            ServiceCosmosDbConfigurationInfo cosmosDbConfigurationInfo = new ServiceCosmosDbConfigurationInfo(offerThroughput);
+            ServiceAuthenticationConfigurationInfo authenticationConfigurationInfo = new ServiceAuthenticationConfigurationInfo(authority, audience, smartOnFhirEnabled);
+
+            var serviceProperties = new ServicesProperties(accessPolicies, provisioningState, cosmosDbConfigurationInfo, authenticationConfigurationInfo);
+
+            return serviceProperties;
+        }
+
+        public static string GetLocationFromProvider(this ResourceManagementClient resourceManagementClient)
```
Comment (YazanMSFT) [inline src/SDKs/HealthcareApis/HealthcareApis.Tests/Helpers/HealthcareApisManagementTestUtilities.cs]: This isn't used

File: src/SDKs/HealthcareApis/HealthcareApis.Tests/Helpers/HealthcareApisManagementTestUtilities.cs [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+            }
+            else
+            {
+                handler.IsPassThrough = true;
+                ResourceManagementClient resourcesClient = context.GetServiceClient<ResourceManagementClient>(handlers: handler);
+                return resourcesClient;
+            }
+        }
+
+        public static string CreateHealthcareApisServicesAccount(HealthcareApisManagementClient healthcareApisManagementClient, string rgname, string kind = null)
```
Comment (YazanMSFT) [inline src/SDKs/HealthcareApis/HealthcareApis.Tests/Helpers/HealthcareApisManagementTestUtilities.cs]: Not being used

File: src/SDKs/HealthcareApis/HealthcareApis.Tests/Helpers/RecordedDelegatingHandler.cs [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+        private HttpResponseMessage _response;
+
+        public RecordedDelegatingHandler()
+        {
+            StatusCodeToReturn = HttpStatusCode.Created;
+            SubsequentStatusCodeToReturn = StatusCodeToReturn;
+        }
+
+        public RecordedDelegatingHandler(HttpResponseMessage response)
```
Comment (YazanMSFT) [inline src/SDKs/HealthcareApis/HealthcareApis.Tests/Helpers/RecordedDelegatingHandler.cs]: Not being used

File: src/SDKs/HealthcareApis/HealthcareApis.Tests/Helpers/RecordedDelegatingHandler.cs [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+        public HttpMethod Method { get; private set; }
+
+        public Uri Uri { get; private set; }
+
+        public bool IsPassThrough { get; set; }
+
+        private int counter;
+
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, System.Threading.CancellationToken cancellationToken)
```
Comment (YazanMSFT) [inline src/SDKs/HealthcareApis/HealthcareApis.Tests/Helpers/RecordedDelegatingHandler.cs:41]: VS says this isn't used either. Is this from a separate class they provide or something?

File: src/SDKs/HealthcareApis/HealthcareApis.Tests/Tests/HealthcareApisAccountTests.cs [hunk 8]
```diff
@@ -0,0 +1,245 @@
...
+                var resourcesClient = HealthcareApisManagementTestUtilities.GetResourceManagementClient(context, handler);
+                var healthCareApisMgmtClient = HealthcareApisManagementTestUtilities.GetHealthcareApisManagementClient(context, handler);
+
+                // Create resource group
+                var rgname = HealthcareApisManagementTestUtilities.CreateResourceGroup(resourcesClient);
+
+                // Generate account name
+                string accountName = TestUtilities.GenerateName("hca");
+
+                var serviceDescription = HealthcareApisManagementTestUtilities.GetServiceDescription();
```
Comment (YazanMSFT) [inline src/SDKs/HealthcareApis/HealthcareApis.Tests/Tests/HealthcareApisAccountTests.cs]: Should this be GetServiceDescriptionWithProperties?

File: src/SDKs/_metadata/HealthcareApis.txt [hunk 9]
```diff
@@ -0,0 +1,14 @@
+﻿Installing AutoRest version: latest
```
Comment (YazanMSFT) [inline src/SDKs/_metadata/HealthcareApis.txt]: Is this file needed?

File: src/SDKs/HealthcareApis/HealthcareApis.Tests/Tests/HealthcareApisAccountTests.cs [hunk 10]
```diff
@@ -0,0 +1,245 @@
...
+
+            using (MockContext context = MockContext.Start(this.GetType().Name))
+            {
+                var resourcesClient = HealthcareApisManagementTestUtilities.GetResourceManagementClient(context, handler);
+                var healthCareApisMgmtClient = HealthcareApisManagementTestUtilities.GetHealthcareApisManagementClient(context, handler);
+
+                // Create resource group
+                var rgname = HealthcareApisManagementTestUtilities.CreateResourceGroup(resourcesClient);
+
+                var serviceAccountsLists = healthCareApisMgmtClient.Services.ListByResourceGroupWithHttpMessagesAsync(rgname);
```
Comment (YazanMSFT) [inline src/SDKs/HealthcareApis/HealthcareApis.Tests/Tests/HealthcareApisAccountTests.cs]: Should we test it actually returning results? The Maps team has a test that checks for empty, then does crestes and checks the properties on the return.

File: src/SDKs/HealthcareApis/HealthcareApis.Tests/SessionRecords/HealthcareApisTests/HealthcareApisListAccountsTest.json [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+      "RequestHeaders": {
+        "x-ms-client-request-id": [
+          "74fad5d2-1e3c-4e30-a0b3-fcf319969fe0"
+        ],
+        "accept-language": [
+          "en-US"
+        ],
+        "User-Agent": [
+          "FxVersion/4.6.25009.03",
+          "Microsoft.Azure.Management.Billing.BillingManagementClient/1.2.0-preview"
```
Comment (YazanMSFT) [inline src/SDKs/HealthcareApis/HealthcareApis.Tests/SessionRecords/HealthcareApisTests/HealthcareApisListAccountsTest.json]: I think this is wrong :P

File: src/SDKs/HealthcareApis/HealthcareApis.Tests/Helpers/RecordedDelegatingHandler.cs (hunk 7 above)
Comment (poadhikagithub) [reply src/SDKs/HealthcareApis/HealthcareApis.Tests/Helpers/RecordedDelegatingHandler.cs:41]: Its internally being used. Not sure why VS is saying not used.
